ruby/ql/lib/codeql/ruby/frameworks/Nokogiri.qll

```diff
@@ -0,0 +1,54 @@
+/**
+ * Provides modeling for `nokogiri`, an XML library for Ruby.
+ */
+
+private import codeql.ruby.ApiGraphs
+private import codeql.ruby.dataflow.FlowSummary
+private import codeql.ruby.Concepts
+
+/**
+ * Provides modeling for `nokogiri`, an XML library for Ruby.
+ */
+module Nokogiri {
+  /**
+   * Flow summary for `nokogiri`. Wraps a string, parsing it as an XML document.
+   */
+  private class XMLSummary extends SummarizedCallable {
+    XMLSummary() { this = "Nokogiri::XML.parse" }
+
+    override MethodCall getACall() { result = any(NokogiriXmlParserCall p).asExpr().getExpr() }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "Argument[0]" and output = "ReturnValue" and preservesValue = false
+    }
+  }
+
+  /** A call that parses XML. */
+  private class NokogiriXmlParserCall extends DataFlow::CallNode {
```

Comment:
Should this extend `XmlParserCall::Range`?

Reply:
Hi Alex! Thanks for the review, about Nokogiri I'm not sure what to do. Since this parser call is a "tiny" version of the parser call written in `XmlParsing.qll` I don't know if I can somehow import it or just copy that part to `Nokogiri.qll`